src/server/blocking_controller.cc

```diff
@@ -214,6 +178,8 @@ void BlockingController::NotifyPending() {
       DVLOG(1) << "Processing awakened key " << sv_key;
       auto w_it = wt.queue_map.find(sv_key);
       if (w_it == wt.queue_map.end()) {
+        // This should not happen because we remove keys from awakened_keys every type we remove
+        // the entry from queue_map. TODO: to make it a CHECK after Dec 2024
```

Comment:
I foresee us totally missing this line in december 🤣

Reply:
we can grep TODOs from time to time and Dec is purely directional :)